src/proof/veriT/veriT_post_processor.cpp

```diff
@@ -0,0 +1,1163 @@
+    // proof term: (or F_1 ... F_i-1 F_i+1 ... F_n G_1 ... G_j-1 G_j+1 ... G_m)
+    // premises: (P1:(or F_1 ... F_i-1 F_i F_i+1 ... F_n),
+    //            P2:(or G_1 ... G_j-1 G_j G_j+1 ... G_m))
+    // args: ()
+    case PfRule::RESOLUTION:{
+      return addVeriTStep(res,VeriTRule::RESOLUTION,children,{},*cdp);
+    }
+    // ======== Chain Resolution
+    // Children: (P1:(or F_{1,1} ... F_{1,n1}), ..., Pm:(or F_{m,1} ... F_{m,nm}))
+    // Arguments: (L_1, ..., L_{m-1})
+    // ---------------------
+    // Conclusion: C_m'
+    // where
+    //   let "C_1 <>_l C_2" represent the resolution of C_1 with C_2 with pivot l,
+    //   let C_1' = C_1 (from P_1),
+    //   for each i > 1, C_i' = C_i <>_L_i C_{i-1}'	
+    //
+    // proof rule: resolution
+    // proof term: (C_m')
+    // premises: (P1:(or F_{1,1} ... F_{1,n1}), ..., Pm:(or F_{m,1} ... F_{m,nm}))
+    // args: ()
+    case PfRule::CHAIN_RESOLUTION:{
+      return addVeriTStep(res,VeriTRule::RESOLUTION,children,{},*cdp);
+    }
+    // ======== Factoring
+    // Children: (P:C1)
+    // Arguments: ()
+    // ---------------------
+    // Conclusion: C2
+    // where
+    //  Set representations of C1 and C2 is the same and the number of literals in
+    //  C2 is smaller than that of C1
+    //
+    //  proof rule: duplicate_literals
+    //  proof term: (C2)
+    //  premises: (P:C1)
+    //  args: ()
+    case PfRule::FACTORING:{
+      return addVeriTStep(res,VeriTRule::DUPLICATE_LITERALS,children,{},*cdp);
+    }  
+    // ======== Reordering
+    // Children: (P:C1)
+    // Arguments: (C2)
+    // ---------------------
+    // Conclusion: C2
+    // where
+    //  Set representations of C1 and C2 is the same but the number of literals in
+    //  C2 is the same of that of C1
+    //
+    // proof rule: 
+    // proof term:
+    // premises:
+    // args: 
+    case PfRule::REORDERING:{
+      //TODO skip
+    }  
+    // ======== Split
+    // Children: none
+    // Arguments: (F)
+    // ---------------------
+    // Conclusion: (or F (not F))
+    //
+    // proof rule: not_not
+    // proof term: (VP1:(or (not (not (not F))) F))
+    // premises: () 
+    // args: ()
+    //
+    // proof rule: not_not
+    // proof term: (VP2:(or (not (not (not (not F)))) (not F))
+    // premises: ()
+    // args: ()
+    //
+    // proof rule: resolution
+    // proof term: (or F (not F))
+    // premises: VP1 VP2
+    // args: ()
+    case PfRule::SPLIT:{
+      Node vp1 = d_nm->mkNode(kind::OR,(d_nm->mkNode(kind::NOT,d_nm->mkNode(kind::NOT,d_nm->mkNode(kind::NOT,args[0]))),args[0]));
+      Node vp2 = d_nm->mkNode(kind::OR,(d_nm->mkNode(d_nm->mkNode(kind::NOT,d_nm->mkNode(kind::NOT,d_nm->mkNode(kind::NOT,args[0]))),d_nm->mkNode(kind::NOT,args[0])))); 
+      
+      return addVeriTStep(vp2,VeriTRule::NOT_NOT,{},{},*cdp) 
+	     && addVeriTStep(vp1,VeriTRule::NOT_NOT,{},{},*cdp) 
+             && addVeriTStep(res,VeriTRule::RESOLUTION,{vp1,vp2},{},*cdp);
+    }
```

Comment:
Kind of beautiful! I like how we derive excluded-middle via double-negation elimination. :) Made me remember of starting to learn logic.